indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -439,6 +439,13 @@ public void handle()
             );
             return;
           }
+          if (SupervisorStateManager.BasicState.IDLE.equals(getState())) {
```

Comment:
Since supervisor state is an enum, we can use `==` to check for equality:
```suggestion
          if (SupervisorStateManager.BasicState.IDLE == getState()) {
```

Reply:
ah, looking into the code it seems like there is `stateManager.isIdle()` that does the same check?

Reply:
I used `getState()` because I can override this in the test to always return IDLE